stan/math/prim/prob/exp_mod_normal_cdf.hpp

```diff
@@ -48,59 +48,55 @@ return_type_t<T_y, T_loc, T_scale, T_inv_scale> exp_mod_normal_cdf(
-        return ops_partials.build(0.0);
-      }
+  for (size_t n = 0, size_y = size(y); n < size_y; n++) {
+    if (is_inf(y_vec[n]) && y_vec[n] < 0.0) {
+      return ops_partials.build(0.0);
     }
+  }
 
+  for (size_t n = 0; n < N; n++) {
     const T_partials_return y_dbl = value_of(y_vec[n]);
     const T_partials_return mu_dbl = value_of(mu_vec[n]);
     const T_partials_return sigma_dbl = value_of(sigma_vec[n]);
     const T_partials_return lambda_dbl = value_of(lambda_vec[n]);
     const T_partials_return u = lambda_dbl * (y_dbl - mu_dbl);
     const T_partials_return v = lambda_dbl * sigma_dbl;
-    const T_partials_return v_sq = v * v;
-    const T_partials_return v_over_sqrt_two = v / SQRT_TWO;
+    const T_partials_return v_over_sqrt_two = v * INV_SQRT_TWO;
+    const T_partials_return inv_sigma = 1 / sigma_dbl;
     const T_partials_return scaled_diff
-        = (y_dbl - mu_dbl) / (SQRT_TWO * sigma_dbl);
+        = (y_dbl - mu_dbl) * INV_SQRT_TWO * inv_sigma;
     const T_partials_return scaled_diff_sq = scaled_diff * scaled_diff;
     const T_partials_return erf_calc
         = 0.5 * (1 + erf(-v_over_sqrt_two + scaled_diff));
-    const T_partials_return deriv_1
-        = lambda_dbl * exp(0.5 * v_sq - u) * erf_calc;
+    const T_partials_return exp_term = exp(0.5 * square(v) - u);
+
+    const T_partials_return deriv_1 = lambda_dbl * exp_term * erf_calc;
     const T_partials_return deriv_2
-        = SQRT_TWO_OVER_SQRT_PI * 0.5
-          * exp(0.5 * v_sq - square(scaled_diff - v_over_sqrt_two) - u)
-          / sigma_dbl;
+        = SQRT_TWO_OVER_SQRT_PI * 0.5 * exp_term
+          * exp(-square(scaled_diff - v_over_sqrt_two)) * inv_sigma;
     const T_partials_return deriv_3
-        = SQRT_TWO_OVER_SQRT_PI * 0.5 * exp(-scaled_diff_sq) / sigma_dbl;
+        = SQRT_TWO_OVER_SQRT_PI * 0.5 * exp(-scaled_diff_sq) * inv_sigma;
 
-    const T_partials_return cdf_n = 0.5 * (1 + erf(u / (v * SQRT_TWO)))
-                                    - exp(-u + v_sq * 0.5) * (erf_calc);
+    const T_partials_return cdf_n
+        = 0.5 * (1 + erf(u / (v * SQRT_TWO))) - exp_term * erf_calc;
 
     cdf *= cdf_n;
 
     if (!is_constant_all<T_y>::value) {
       ops_partials.edge1_.partials_[n] += (deriv_1 - deriv_2 + deriv_3) / cdf_n;
     }
     if (!is_constant_all<T_loc>::value) {
-      ops_partials.edge2_.partials_[n]
-          += (-deriv_1 + deriv_2 - deriv_3) / cdf_n;
+      ops_partials.edge2_.partials_[n] -= (deriv_1 - deriv_2 + deriv_3) / cdf_n;
     }
     if (!is_constant_all<T_scale>::value) {
       ops_partials.edge3_.partials_[n]
-          += (-deriv_1 * v - deriv_3 * scaled_diff * SQRT_TWO
-              - deriv_2 * sigma_dbl * SQRT_TWO
-                    * (-SQRT_TWO * 0.5
-                           * (-lambda_dbl + scaled_diff * SQRT_TWO / sigma_dbl)
-                       - SQRT_TWO * lambda_dbl))
+          -= (deriv_1 * v + deriv_3 * scaled_diff * SQRT_TWO
+              - deriv_2 * (v + SQRT_TWO * scaled_diff))
```

Comment:
Dope! I'm glad there are tests, but I also validated this by careful symbolic manipulation. Thanks for simplifying.